spec/support/auth_helper.rb

```diff
@@ -5,39 +5,29 @@ def http_login(username = 'username', password = 'password')
         request.env['HTTP_AUTHORIZATION'] = ActionController::HttpAuthentication::Basic.encode_credentials(username, password)
       end
 
-      def login_as(user)
+      def login_as(user, stub_controller: false)
         User.current_user = user
+        puts "WARNING: double stubbing user - only use login_as or stub_user once" if user != User.current_user
         session[:userid]  = user.userid
         session[:group]   = user.current_group_id
+        allow(controller).to receive(:current_user).and_return(user) if stub_controller
         user
       end
 
-      # TODO: Stub specific features, document use
-      def stub_user(features:, user: FactoryGirl.build(:user_with_group))
-        allow(controller).to receive(:current_user).and_return(user)
-        allow(User).to receive(:current_user).and_return(user)
+      # Stubs the user in context for a controller
+      # @param features (:all|:none|Array<>,String,Symbol) features for a user
```

Comment:
when specific features are chosen, how are they passed...as Strings?  I see Array of symbol, but I would have expected String identifiers.

That being said, there already is an "everything" product feature, so we should deprecate `:all`, and none can be handled by passing an empty Array, so we should probably deprecate `:none` as well.  Then there are no "special" feature names, only real ones.

Reply:
thnx - strings or symbols work

Reply:
a) I would like to `s/all/everything` in a future PR. it is in many repos.
b) `:none` seems to work very well. It creates all the joining models. The user ends up with feature(s), just features that aren't ones that affect us